### PR TITLE
persist the libp2p keypair

### PR DESCRIPTION
**Pull request by tiainen**
## Description

Fixes pyrsia/pyrsia#505

Stores the keypair that is used for the p2p network in the pyrsia artifact path.

## PR Checklist

- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/good_pr.md)
- [x] I've included a good title and brief description along with how to review them.
- [x] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).
- [x] I've requested a review  from "pyrsia/collaborators"

### Code Contributions

- [x] I've built the code `cargo build --workspace`.
- [x] I've run the unit tests `cargo test --workspace`.

### CODE REVIEW

File: src/network/p2p.rs [hunk 1]
```diff
@@ -48,13 +49,13 @@ use std::iter;
 ///  * [`EventLoop`]: used to process events received from the p2p [`Swarm`] and commands from the p2p [`Client`]
 pub async fn create_components(
 ) -> Result<(Client, impl Stream<Item = Event>, EventLoop), Box<dyn Error>> {
-    let local_keys = identity::Keypair::generate_ed25519();
+    let local_keypair = keypair_util::create_ed25519();
```
Comment (prince-chrismc) [inline src/network/p2p.rs:52]: "create" does not really capture the magic this function is doing, maybe this?

```suggestion
    let local_keypair = keypair_util::initiate_ed25519();
```

**Review by prince-chrismc — COMMENTED** [review]
One tiny suggestions but otherwise LGTM ✅

File: src/network/p2p.rs (hunk 1 above)
Comment (tiainen) [reply src/network/p2p.rs:52]: I had indeed some trouble to find a good name. This is what the function does: load a key from file, if it's not there, create a new one and save that one for future calls. `initiate` makes it sound like it creates a new keypair each time it's called. Would `load_or_create_ed25519` be any better?

Comment (johanvos) [reply src/network/p2p.rs:52]: I agree create is misleading and initiate is vague, so +1 from me for load_or_create

**Review by johanvos — APPROVED** [review]

File: src/util.rs [hunk 2]
```diff
@@ -15,3 +15,4 @@
 */
 
 pub mod env_util;
+pub mod keypair_util;
```
Comment (betarelease) [inline src/util.rs:18]: my only suggestion would be to not add stuff to utils. In fact we should figure out how to move env_util into its own concern.

File: src/util/keypair_util.rs [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        KEYPAIR_FILENAME
+    )
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn load_non_existing_keypair_fails() {
```
Comment (betarelease) [inline src/util/keypair_util.rs:85]: good test coverage. Thanks for the diligence in covering these cases.

**Review by betarelease — APPROVED** [review]
Once you address my comments feel free to merge. Ping me if you need another review and I can quickly verify.

File: src/util.rs (hunk 2 above)
Comment (tiainen) [reply src/util.rs:18]: Could you create an issue for this, including the rationale for why you believe having a util section is bad practice?

**Pull request merged**